cmd/state-exec/meta.go

```diff
@@ -34,10 +37,24 @@ func newExecutorMeta(execPath string) (*executorMeta, error) {
 		return nil, fmt.Errorf("cannot get matching bin by path: %w", err)
 	}
 
+	transformEnv := true
+	if projectDir := os.Getenv(constants.ActivatedStateEnvVarName); projectDir != "" {
+		// Note: cannot import runtime/target for ProjectDirToTargetDir() because that would bloat
+		// up the exe from <4MB to >15MB.
+		projectCacheDir := filepath.Join(storage.CachePath(), hash.ShortHash(projectDir))
+		if projectCacheDir == execDir {
+			transformEnv = false // runtime environment variables already set
+		}
+	}
+
+	env := os.Environ()
+	if transformEnv {
+		env = transformedEnv(env, meta.Env)
+	}
```

Comment:
nit: Consider moving the conditional into a new function (e.g. "shouldTransformEnv"), then calling as `env := os.Environ(); if shouldTransformEnv() {; env = transformedEnv(env, meta.Env);}")`.